src/materials/DamagePlasticityStressUpdate.C

```diff
@@ -0,0 +1,680 @@
+  params.addRangeCheckedParam<Real>("yield_strength_in_tension",
+                                    "yield_strength_in_tension >= 0",
+                                    "Tensile yield strength of concrete");
+  params.addRangeCheckedParam<Real>("fracture_energy_in_tension",
+                                    "fracture_energy_in_tension >= 0",
+                                    "Fracture energy of concrete in uniaxial tension");
+
+  params.addRangeCheckedParam<Real>("yield_strength_in_compression",
+                                    "yield_strength_in_compression >= 0",
+                                    "Absolute yield compressice strength");
+  params.addRequiredParam<Real>("compressive_damage_at_max_compressive_strength",
+                                "damage at maximum compressive strength");
+  params.addRequiredParam<Real>("maximum_strength_in_compression",
+                                "Absolute maximum compressive strength");
+  params.addRangeCheckedParam<Real>("fracture_energy_in_compression",
+                                    "fracture_energy_in_compression >= 0",
+                                    "Fracture energy of concrete in uniaxial compression");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "tip_smoother",
+      "tip_smoother>=0",
+      "Smoothing parameter: the cone vertex at mean = cohesion*cot(friction_angle), will be "
+      "smoothed by the given amount. Typical value is 0.1*cohesion");
+  params.addParam<bool>("perfect_guess",
+                        true,
+                        "Provide a guess to the Newton-Raphson proceedure "
+                        "that is the result from perfect plasticity.  With "
+                        "severe hardening/softening this may be "
+                        "suboptimal.");
+  params.addClassDescription("Damage Plasticity Model for concrete");
+  return params;
+}
+
+DamagePlasticityStressUpdate::DamagePlasticityStressUpdate(const InputParameters & parameters)
+  : MultiParameterPlasticityStressUpdate(parameters, 3, 1, 2),
+    _f_tol(getParam<Real>("yield_function_tol")),
+
+    _alfa(getParam<Real>("biaxial_uniaxial_compressive_stress_factor")),
+    _alfa_p(getParam<Real>("dilatancy_factor")),
+    _s0(getParam<Real>("stiff_recovery_factor")),
+
+    _Chi(getParam<Real>("ft_ep_slope_factor_at_zero_ep")),
+    _Dt(getParam<Real>("tensile_damage_at_half_tensile_strength")),
+    _ft(getParam<Real>("yield_strength_in_tension")),
+    _FEt(getParam<Real>("fracture_energy_in_tension")),
+
+    _fyc(getParam<Real>("yield_strength_in_compression")),
+    _Dc(getParam<Real>("compressive_damage_at_max_compressive_strength")),
+    _fc(getParam<Real>("maximum_strength_in_compression")),
+    _FEc(getParam<Real>("fracture_energy_in_compression")),
+
+    _at(1.5 * std::sqrt(1 - _Chi) - 0.5),
+    _ac((2. * (_fc / _fyc) - 1. + 2. * std::sqrt(std::pow((_fc / _fyc), 2.) - _fc / _fyc))),
```

Comment:
Integer power should use Utility::pow<2>()

Reply:
Addressed